l2geth/rollup/sync_service.go

```diff
@@ -632,101 +486,171 @@ func (s *SyncService) SetLatestVerifiedIndex(index *uint64) {
 	}
 }
 
-// reorganize will reorganize to directly to the index passed in.
-// The caller must handle the offset relative to the ctc.
-func (s *SyncService) reorganize(index uint64) error {
-	if index == 0 {
-		return nil
-	}
-	err := s.bc.SetHead(index)
-	if err != nil {
-		return fmt.Errorf("Cannot reorganize in syncservice: %w", err)
-	}
+// GetLatestBatchIndex reads the last processed transaction batch
+func (s *SyncService) GetLatestBatchIndex() *uint64 {
+	return rawdb.ReadHeadBatchIndex(s.db)
+}
 
-	// TODO: make sure no off by one error here
-	s.SetLatestIndex(&index)
+// GetNextBatchIndex reads the index of the next transaction batch to process
+func (s *SyncService) GetNextBatchIndex() uint64 {
+	index := s.GetLatestBatchIndex()
+	if index == nil {
+		return 0
+	}
+	return *index + 1
+}
```

Comment:
Can probably remove this function? Returning `index+ 1` from a state variable feels a bit weird, since at the only place where we use it, we `- 1` it. I think the `index == nil` check could go inside the GetLatestBatchIndex function.

Reply:
The `GetNext...` functions are now all used and are meant to abstract away the `nil` checks since they are done fairly often